### PR TITLE
SWP-557755 Delay calls to GH search api to avoid hitting limits

### PR DESCRIPTION
**Pull request by markelarizaga**
[SWP-55775](https://splashthat.atlassian.net/browse/SWP-55775)

- This PR introduces a delay between the calls to the GitHub's Search end point in order to avoid hitting the API limit.
- Also, it brings the code of 2 dependencies of the original project's code into the project, in order not to have to deal with 3 projects to make changes in this action, and avoid the need to publish NPM packages of those dependencies.
- It also increases the logs thrown to the GitHub Actions page in order to better show what's going on.
This PR has a [counterpart in the frontend project](https://github.com/SplashThat/splash-frontend/pull/244).

### CODE REVIEW

File: lib/detect.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+const Octokit = require('@octokit/rest');
```
Comment (markelarizaga) [inline lib/detect.js:1]: In the original repo, this file was an independent project which we had to publish to NPM if we wanted to use as a separate thing. I decided to merge all the code within a single project for ease of use.

File: lib/git-pr-release.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
+const detect = require('./detect');
```
Comment (markelarizaga) [inline lib/git-pr-release.js:1]: Same as above, I merged this code from a separate project into this one.

File: lib/detect.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  // To avoid 422 error -- 'The search is longer than 256 characters',
+  // Divide 'pulls' to chunks.
+  const chunkedCommits = toChunk(merge_commits, 20);
+  core.info(`Created an array of ${chunkedCommits.length} chunks`);
+  const pulls = await chunkedCommits.reduce(async (accum, chunk, index) => {
+    // Awaiting accum because the function passed to reduce is async, therefore returns a promise
+    const acc = await accum;
+    // Avoid making too many calls too fast to keep below GitHub's search rate limit
+    // This will cause the calls to be made with a 2 seconds separation
+    const timeToSleep = (index + 1) * 2000;
```
Comment (markelarizaga) [inline lib/detect.js]: Introduces a 2 second delay between calls to avoid the API limit.

Comment (carlesnunez) [reply lib/detect.js]: Nitpick but this could be a constant

**Review by carlesnunez — APPROVED** [review]

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "git-pr-release-action",
-  "version": "1.0.0",
+  "name": "@splashthat/git-pr-release-action",
+  "version": "1.0.2",
```
Comment (ivanrod) [inline package.json:3]: Why?

**Review by ivanrod — APPROVED** [review]
This is great!
Not a big fan of creating forks that we will need to maintain but in this case it looks totally reasonable to me!

File: package.json (hunk 4 above)
Comment (markelarizaga) [reply package.json:3]: I guess you miss `1.0.1`. The reason why it doesn't appear in this PR is that I made a version `1.0.1` in the branch, but later I bumped the version again to `1.0.2`.
If you see the repo, you will still be able to find the tag `v1.0.1`.

**Pull request merged**